### ISSUE TITLE
Sync with Cookie deletion and "session" usage?

### ISSUE THREAD

**Issue by ghost** [dillbyrne/random-agent-spoofer#72]
I am wondering how much use this plugin is if cookies aren't deleted at the same moment that UA is switched? I am using "self-destructing cookies" extension (https://addons.mozilla.org/en-US/firefox/addon/self-destructing-cookies/?src=api) but I think both extensions should somehow cooperate.

Also, switching after random time is nice but it might be very interesting to have switching based on browsing activity such as opening & closing tabs and session characteristics. This is not trivial.. I think it would be nasty to have the UA switched and cookies deleted in the middle of an Ebay purchase but might still be a good idea in other circumstances.

**Comment by Mosrite** [comment]
Couldn't you just add a feature to change the profile for every new tab that is opened? By this, the functionality would be more similar to the SDC addon.

**Comment by dillbyrne** [comment]
@Mosrite This would then change all the currently opened tabs to use that new profile once a request was made so it is in effect the same as having it on a timer.

**Comment by dillbyrne** [comment]
I intend to begin work on this in the release after 0.9.5.4.  In addition to aiming for Self destructing cookie functionality, it should block _ga (google analytics) and _gat (google analytics tracker) cookies by default.

**Comment by sueridgepipe** [comment]
I don't see the point of including tracker cookie blocking in RAS where there are other addons that do it much more comprehensively (ie Ghostery, UBlock Origin, uMatrix, etc).

Self-Destructing Cookies addon works perfectly for me, not sure how it would be effectively automated in RAS.  How would RAS handle different cookies across multiple tabs when profiles change?  How would this differ from SDC handles cookie deletion?

Profile spoofing and fingerprint protection is what RAS does best, don't over complicate it with functionality already handled in other addons IMHO.

Instead of bloating RAS unnecessarily with redundant/replicated functionality why not just create a best practices section to the wiki?  This could contain opinions on the best combo of addons (in addition to RAS) to achieve maximum privacy and make users most difficult to track.

For example, BetterPrivacy addon is a must for automatic removal of persistent Flash cookies, and HSTS cookies are avoided via use of private browsing.

Just my 2c worth....

**Comment by rgh36167** [comment]
@sueridgepipe : profile spoofing and fingerprint protection are completely useless if you forget to delete cookies at exactly the moment that you switch profiles. Neither SDC nor any other extensions that you mention can do it unless they cooperate with RAS to synchronise their action in some way.

**Comment by sueridgepipe** [comment]
So what happens when you have multiple tabs open, each logged into a different service requiring cookies, and the profile spoofs every 1-5 minutes?  Will RAS delete all the cookies on each profile change?  Will this not break the site(s)?  If these cookies are not deleted, why would this break profile spoofing?  Is profile and fingerprinting data stored in the cookies?  If so, why would that matter as it is a spoofed profile anyway?

I never have cookies enabled unless I am logging into a service that requires it (ie banking, online store, forums, etc).  Once I am done I disable cookies again and ensure all cookies are deleted (SDC).  For services that require cookies I am never logged into more than one at a time.  I manually ensure cookies do not persist, ever, and am vigilant about it.

I use the same process for sites that require Flash, which is never enabled unless I need it.  Once done on that site I disable Flash again and exit to ensure persistent Flash supercookies are deleted (via BetterPrivacy addon).

I only enable Javascript when absolutely necessary too, and even then I limit the scripts it can run using uMatrix.

I use the QuickJava addon every single day to quickly and easily enable/disable browser functionality like Javascript, Cookies, Flash, Java, etc within FF with a single click.  My rule is everything is disabled by default unless it breaks something on site I am viewing / using.

Basically the point I am trying to make in explaining my browsing habits is that how you browse online is nearly as important as the addons you install in terms of privacy.

**Comment by rgh36167** [comment]
@sueridgepipe : in this situation random profile switching will harm you. Of course as long as the cookie is preserved the server will know it is still you, no matter how you switch RAS profile the cookie will identify you. This way the evil server can collect all RAS fingerprints tied to your identity quite easily and knows that you use RAS.

It comes even worse. Consider what happens when you are logged into a legit banking site with some cookies and switch RAS fingerprints. The banking server notices that suddenly the same cookie(s) is used by a seemingly different browser and will suspect that your online-banking session has been hijacked, MITMed or your computer compromised. At the very least the server will terminate the session and quite possibly disable your online banking until you can prove to your bank that you have cleaned up your computer from malware.

I think doing both cookie deletion & agent switching correctly by hand is nearly impossible. It is also quite hard to do right for an extension but RAS has the goal to at least try this.

What could be done is using only private tabs and switch user agent for each private tab separately. When going from one domain to another within a tab there should be an option to switch or keep cookies & browser profile. Or maybe both should be switched by default on every transition unless in some whitelist (ebay&paypal come to mind where it would be better to preserve fingerprint and cookies)

Even without javascript and cookies html still offers infinite possibilities to track you (tracking ids in links, ping, hidden fields..). It may be therefore better for privacy not to block cookies but make sure these are frequently deleted - at the right time.
